tempest/api/workloadmgr/barbican/test_barbican.py

```diff
@@ -3120,3 +3134,215 @@ def test_10_barbican(self):
+    @decorators.attr(type='workloadmgr_cli')
+    def test_11_barbican(self):
+        try:
+            reporting.add_test_script(str(__name__) + \
+                    "_validate_scheduler_trust_with_scheduler_enabled")
+            if self.exception != "":
+                LOG.error("pre req failed")
+                raise Exception(str(self.exception))
+            LOG.debug("pre req completed")
+            global vm_id
+            global volume_id
+            global exception
+            global secret_uuid
+            vm_id = self.vm_id
+            volume_id = self.volume_id
+            exception = self.exception
+
+            # Create scheduled encrypted workload
+            self.secret_uuid = self.create_secret(secret_cleanup=False)
+            secret_uuid = self.secret_uuid
+
+            self.start_date = time.strftime("%m/%d/%Y")
+            self.start_time = time.strftime("%I:%M %p")
+            self.wid = self.workload_create([vm_id], tvaultconf.parallel,
+                        jobschedule={"start_date": self.start_date,
+                            "start_time": self.start_time,
+                            "interval": tvaultconf.interval,
+                            "retention_policy_type":
+                                tvaultconf.retention_policy_type,
+                            "retention_policy_value":
+                                tvaultconf.retention_policy_value,
+                            "enabled": "True"}, encryption=True,
+                        secret_uuid=secret_uuid)
+            LOG.debug("Workload ID: " + str(self.wid))
+            self.wait_for_workload_tobe_available(self.wid)
+            if(self.getWorkloadStatus(self.wid) == "available"):
+                reporting.add_test_step(
+                    "Create scheduled encrypted workload", tvaultconf.PASS)
+            else:
+                raise Exception("Create scheduled encrypted workload")
+
+            #Execute scheduler-trust-validate CLI command
+            self._execute_scheduler_trust_validate_cli()
+
+            #Fetch trust list from API
+            trust_list = self.get_trusts()
+
+            #Verify if trust details returned in steps 3 and 4 match
+            found = False
+            for trust in trust_list:
+                if trust['name'] == self.wlm_trust['name']:
+                    found = True
+                    break
+            if found and self.wlm_trust_valid and self.wlm_scheduler:
+                reporting.add_test_step("Verify valid trust", tvaultconf.PASS)
+            else:
+                reporting.add_test_step("Verify valid trust", tvaultconf.FAIL)
+                reporting.set_test_script_status(tvaultconf.FAIL)
+
+            #Delete the trust using API
+            if self.delete_trust(self.wlm_trust['name']):
+                reporting.add_test_step("Delete trust", tvaultconf.PASS)
+            else:
+                raise Exception("Delete trust")
```

Comment:
Do we need to mark test case as FAIL before raising an exception here for delete trust ?

Reply:
In except block, we are adding the test step with FAIL status as well as setting the test script status to FAIL. Hence just raising the exception, takes care of it automatically.